Gems/WhiteBox/Code/Source/Components/WhiteBoxColliderComponent.cpp

```diff
@@ -33,6 +33,11 @@ namespace WhiteBox
         }
     }
 
+    void WhiteBoxColliderComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("PhysicsStaticRigidBodyService"));
```

Comment:
Whitebox gem has a dependency on PhysX, not the other way around. Wouldn't be more appropriate to make whitebox component incompatible with some physics service, instead of making several physics components incompatible with a service added by whitebox?

Sorry if I'm misreading this.

Reply:
I don't think either has a dependency on the other (except that the white box editor tests depend on physx editor). The white box collider does everything it needs to through the AzFramework physics API, so it doesn't need to know about a specific physics backend. So I'd like to not add any reference to PhysX into the white box gem.

Reply:
Had a quick chat with @greerdv. The addition of "PhysicsStaticRigidBodyService" here is the right approach, sadly the status of services in PhysX Gem is a bit odd at the moment as they provide "PhysX..." services when they can be "Physics..." services too. And if that were the case this PR would have been only about WhiteBoxColliderComponent providing "PhysicsStaticRigidBodyService" and that's it, no changes to PhysX gem would have been required.

Approving.